tasks/collect.sh

```diff
@@ -1,42 +1,91 @@
 #!/bin/bash
 
-temp_module_directory=/var/tmp/puppet_modules
-output_directory=/var/tmp/health_check_lite
-output_file=$output_directory/health_check_lite.txt
-support_script_output_file=$output_directory/support_script_output.log
+# Variables that need to be interpolated as part of the command won't show up here
+# Should still be useful
+_debug () {
+  echo "DEBUG: running $@" >>"$_tmp.debug"
+}
 
-echo "Collecting HealthCheck Lite"
+declare PT__installdir
+source "$PT__installdir/healthcheck_lite/files/common.sh"
+[[ $PATH =~ "/opt/puppetlabs/bin" ]] || export PATH="/opt/puppetlabs/bin:${PATH}"
 
-mkdir -p $output_directory
-rm -f $output_directory/*.gz
+shopt -s nullglob extglob globstar || fail "This utility requires Bash >=4.0"
+trap '_debug $BASH_COMMAND' DEBUG
 
-echo "Puppet Enterprise HealthCheck Lite: $(date)" > $output_file
-echo >> $output_file
+(( $EUID == 0 )) || fail "This utility must be run as root"
 
-grep -i -v UUID /etc/puppetlabs/license.key >> $output_file 2>/dev/null
-echo >> $output_file
+tmp_dir=/var/tmp/puppet_modules
+output_dir=/var/tmp/health_check_lite
+output_file="$output_dir/health_check_lite.txt"
+support_script_output_file="$output_dir/support_script_output.log"
+# Not currently a parameter
+ticket="${ticket:-HCL}"
 
-/opt/puppetlabs/bin/puppet enterprise support --classifier --dir $output_directory --log-age 3 --ticket HCL > $support_script_output_file 2>&1
+[[ -d $output_dir ]] || {
+  mkdir "$output_dir" || fail "Error creating output directory"
+}
 
-if [ -d $temp_module_directory/pe_tune ]; then
-  echo 'puppet pe tune' >> $output_file
-  puppet pe tune --modulepath $temp_module_directory >> $output_file 2>/dev/null
-  echo >> $output_file
+# Remove any files from previous runs
+find "$output_dir" -mindepth 1 -delete || fail "Error removing previous files"
 
-  echo 'puppet pe tune --current' >> $output_file
-  puppet pe tune --modulepath $temp_module_directory --current >> $output_file 2>/dev/null
-  echo >> $output_file
+# Clone stdout, then redirect it to our output file for the following steps.
+exec 3>&1
+exec >>"$output_file"
+
+echo "Puppet Enterprise HealthCheck Lite: $(date)"
+echo
+
+grep -i -v UUID /etc/puppetlabs/license.key
+
+if puppet enterprise support --help | grep -q -- '--log-age'; then
```

Comment:
This could/should be refactored into a function :)

Reply:
Not sure I agree.  It's just a oneliner, and all the instances of it are grouped together.

Reply:
True, but its repetitive and parameterizable, and as you add checks for `--upload` ...

Reply:
I decided dumping the output of `puppet enterprise support --help` to a file and using a function is better and faster. :+1: